### PR TITLE
feature/PELAGOS-5191-add-anzsrc-keywords-to-metadata

### PR DESCRIPTION
**Pull request by fightingtexasaggie**
Needs PELAGOS-5172 merged into the epic first so we have metadata to generate from a submission with anzsrc keywords.

### CODE REVIEW

File: templates/MetadataGenerator/MD_DataIdentification.xml.twig [hunk 1]
```diff
@@ -65,6 +65,21 @@
             }
         %}
     </gmd:descriptiveKeywords>
+    <gmd:descriptiveKeywords>
+        {% set anzsrc_keywords = [] %}
+        {% for keyword in dataset.datasetSubmission.keywords %}
+            {% if keyword.type.value == 'anzsrc' %}
```
Comment (fightingtexasaggie) [inline templates/MetadataGenerator/MD_DataIdentification.xml.twig:71]: Is there a better way to handle enum constants?

File: templates/MetadataGenerator/ANZSRC_Keywords.xml.twig [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<gmd:MD_Keywords>
+    {% if keywords is empty %}
```
Comment (fightingtexasaggie) [inline templates/MetadataGenerator/ANZSRC_Keywords.xml.twig:2]: redundant. Question for PO: Is it alright to leave this entire section out if we don't have anzsrc keywords? (It's valid without it)

Comment (mickel1138) [reply templates/MetadataGenerator/ANZSRC_Keywords.xml.twig:2]: Yes, leave it out.

File: templates/MetadataGenerator/MD_DataIdentification.xml.twig (hunk 1 above)
Comment (mickel1138) [reply templates/MetadataGenerator/MD_DataIdentification.xml.twig:71]: Yes, the twig constant function: `constant('App\\Enum\\KeywordType::TYPE_ANZSRC') `

Comment (mickel1138) [reply templates/MetadataGenerator/MD_DataIdentification.xml.twig:71]: Also, value is not needed.

File: templates/MetadataGenerator/ANZSRC_Keywords.xml.twig [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            <gmd:title>
+                <gco:CharacterString>ANZSRC Fields of Research</gco:CharacterString>
+            </gmd:title>
+            <gmd:date>
+                <gmd:CI_Date>
+                    <gmd:date>
+                        <gco:Date>2020</gco:Date>
+                    </gmd:date>
+                    <gmd:dateType>
+                        <gmd:CI_DateTypeCode codeList="https://data.noaa.gov/resources/iso19139/schema/resources/Codelist/gmxCodelists.xml#CI_DateTypeCode" codeListValue="revision">revision</gmd:CI_DateTypeCode>
```
Comment (mickel1138) [inline templates/MetadataGenerator/ANZSRC_Keywords.xml.twig:24]: Shouldn't revision be something?

**Review by mickel1138 — APPROVED** [review]
LGTM

**Pull request merged**